pydoc-markdown/src/pydoc_markdown/contrib/renderers/docusaurus.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf8 -*-
+from nr.databind.core import Field
+from nr.interface import implements, override
+from pathlib import Path
+from pydoc_markdown.interfaces import Renderer
+from pydoc_markdown.contrib.renderers.markdown import MarkdownRenderer
+from typing import Any, Dict, List, Text
+import docspec
+import json
+import os.path
+
+
+@implements(Renderer)
+class DocusaurusRenderer(MarkdownRenderer):
+  """
+  Produces Markdown files. This renderer is often used by other renderers, such as
+  #MkdocsRenderer and #HugoRenderer. It provides a wide variety of options to customize
+  the generated Markdown files.
+
+  ### Options
+  """
+
+  #: If enabled, inserts anchors before Markdown headers to ensure that
+  #: links to the header work. This is disabled by default because Docusaurus
+  #: supports this automatically.
+  insert_header_anchors = Field(bool, default=False)
+
+  #: The path where the docusaurus docs content is. Defaults "docs" folder.
+  docs_base_path = Field(str, default='docs')
+
+  #: The output path inside the docs_base_path folder, used to ouput the
+  #: module reference.
+  relative_output_path = Field(str, default='reference')
+
+  #: The sidebar path inside the docs_base_path folder, used to ouput the
+  #: sidebar for the module reference.
+  relative_sidebar_path = Field(str, default='sidebar.json')
+
+  #: The top-level label in the sidebar. Default to 'Reference'.
+  sidebar_top_level_label = Field(str, default='Reference')
+
+  #: Skip documenting modules if empty. Defaults to True.
+  skip_empty_modules = Field(bool, default=True)
+
+  #: Escape html in docstring. Default to True.
+  escape_html_in_docstring = Field(bool, default=True)
+
+  def _render_header(self, fp, level, obj):
+    if isinstance(obj, docspec.Module):
+      fp.write('---\n')
+      fp.write(f'sidebar_label: {obj.name}\n')
+      fp.write(f'title: {obj.name}\n')
+      fp.write('---\n\n')
+      return
+
+    super(DocusaurusRenderer, self)._render_header(fp, level, obj)
```

Comment:
for some reason python3 syntax doesn't work here (I guess because of `@implements(Renderer)`)

Reply:
Yes, unfortunately that is currently a limitation of the `nr.interface.implements` decorator.